### ISSUE TITLE
OpenGDS JAR as release asset

### ISSUE THREAD

**Issue by JohT** [neo4j/graph-data-science#275]
It would be great to get the OpenGDS JAR as release asset additionally to the existing one. 

The steps to do that are already described in the main documentation under [Building the library/Packaging the library](https://github.com/neo4j/graph-data-science#building-the-library). This could be done with GitHub Actions in an automated manner. Of course, this would need to fit/join the current release process (that I'm not aware of).

### Alternatives
An alternative would be to create an own non official repository that clones this repository, creates the bundle and then releases it.

### Additional context
I'm open to contribute GitHub Action(s) if this is of any help.

PS.: Thanks 👏 for this awesome library.

**Comment by FlorentinD** [comment]
Hello @JohT ,
thanks for your request.
Could you explain why you want to use the OpenGDS jar?

**Comment by Mats-SX** [comment]
We don't build and distribute OpenGDS. It's meant to be built manually by whoever wants to modify and use it. If you do not intend to modify GDS I don't see a lot of value of building it. The license is the same either way. The added benefit would be that you can select what commit to build from, instead of following our lead.

You are free to use Neo4j GDS as distributed by us. The license is a prefixed GPLv3, which means you have to license any work that uses or redistributes GDS also using GPLv3 (the prefix only states that a Neo4j commercial agreement supersedes the license; if you pay for our product the terms of the contract will rule above the OSS license). It seems to me that you are already licensing your work using GPLv3 so it shouldn't be an issue.

In order to unlock the Enterprise Edition features (such as unlimited concurrency) one has to buy a Neo4j GDS license from us regardless. This is a built-in restriction. 

Generally, I would recommend to use the GDS that we build and distribute. If your project has a more permissive license, such as APLv2, BSD, MIT, etc which do not conform to the "viral" aspects of the GPLv3 then unfortunately you cannot include either of OpenGDS or Neo4j GDS. Those are the terms of the GPLv3 license; any derivative work has to be made available also under the same license.

So in summary: use GDS with your GPLv3-licensed work. Fork and build your own OpenGDS if you want to modify the software -- there's not much value in building a "vanilla" OpenGDS.

Some resources if you want to read up on OSS licenses:

- https://www.gnu.org/licenses/license-list.en.html
- https://www.apache.org/licenses/GPL-compatibility.html
- https://producingoss.com/en/legal.html
- https://producingoss.com/en/license-choosing.html

**Comment by Mats-SX** [comment]
Disregard my previous reply, it contained inaccuracies (so I deleted it).

**Comment by JohT** [comment]
No worries.

I forgot to mention that I've created the following repository that automates the Open Graph Data Science build, packaging and release: https://github.com/JohT/open-graph-data-science-packaging

It is only a temporary solution and I might be removed in future.

**Comment by Mats-SX** [comment]
Well, I was mistaken on the license we ship with Neo4j GDS. It does indeed not fit for your purpose, so your request (and solution) is valid.
